.planning/research/v1.1-VITE-SOURCEMAPS.md

```
@@ -0,0 +1,106 @@
+# v1.1 Research: Vite/Rolldown sourcemap function-name preservation (issue #19)
+
+> **AMENDED 2026-04-30 — empirical correction from PR #21 verification:**
+> The TL;DR + body below say `keepNames: true` works by injecting a `__name(fn, 'orig')` helper. **Empirically Rolldown does NOT emit that helper** — `grep -c '__name(' dist/assets/*.js` returns 0 even when keepNames is enabled and working. Rolldown's Oxc minifier preserves names by leaving literal `function Name(...)` declarations in the output instead. The fix itself (`build.rolldownOptions.output.keepNames: true`) is correct and the Sentry deploy-preview event at `2026-04-30T09:20:52Z` shows fully un-mangled stack frames — only the assertion mechanism in this doc was wrong. Use `grep -lE 'function (RootLayout\|AppErrorFallback\|RenderThrowSmoke)\b' dist/assets/*.js | wc -l >= 1` (or `jq -r '.names[]' dist/assets/*.js.map`) as the Rolldown-correct mechanical check.
+
+## TL;DR
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)